soc/intel/intel_adsp/ace/multiprocessing.c

```diff
@@ -147,6 +148,11 @@ void soc_start_core(int cpu_num)
 		} else {
 			*rom_jump_vector = (uint32_t) dsp_restore_vector;
 		}
+
+		/* The primary core cannot enter power gating if any of the secondary cores are
+		 * active.
+		 */
+		pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
```

Comment:
So each secondary gets a lock reference and when the last gives up its reference, idle is allowed, right?

Reply:
Exactly.